File: Samples/AdvancedImagingTest/PYTHON/AdvancedImagingTest.py
#---------------------------------------------------------------------------------------
# Copyright (c) 2001-2022 by PDFTron Systems Inc. All Rights Reserved.
# Consult LICENSE.txt regarding license information.
#---------------------------------------------------------------------------------------

import site
site.addsitedir("../../../PDFNetC/Lib")
import sys
from PDFNetPython import *

sys.path.append("../../LicenseKey/PYTHON")
from LicenseKey import *

# Relative path to the folder containing test files.
input_path = "../../TestFiles/AdvancedImaging/"
output_path = "../../TestFiles/Output/"

# ---------------------------------------------------------------------------------------
# The following sample illustrates how to use Advanced Imaging module
# --------------------------------------------------------------------------------------

def main():

    # The first step in every application using PDFNet is to initialize the
    # library and set the path to common PDF resources. The library is usually
    # initialized only once, but calling Initialize() multiple times is also fine.
    PDFNet.Initialize(LicenseKey)

    # The location of the Advanced Imaging Module
    PDFNet.AddResourceSearchPath("../../../PDFNetC/Lib/")

    if not AdvancedImagingModule.IsModuleAvailable():

        print("""
        Unable to run AdvancedImaging2PDFTest: PDFTron SDK Advanced Imaging module not available.
        ---------------------------------------------------------------
        The Advanced Imaging module is an optional add-on, available for download
        at http://www.pdftron.com/. If you have already downloaded this
        module, ensure that the SDK is able to find the required files
        using the PDFNet::AddResourceSearchPath() function.""")

    else:

        inputFileName1 = "xray.dcm"
        outputFileName1 = inputFileName1 + ".pdf"
        doc1 = PDFDoc()
        Convert.FromDICOM(doc1, input_path + inputFileName1, None)
        doc1.Save(output_path + outputFileName1, 0)

        inputFileName2 = "jasper.heic"
        outputFileName2 = inputFileName2 + ".pdf"
        doc2 = PDFDoc()
        Convert.ToPdf(doc2, input_path + inputFileName2)
        doc2.Save(output_path + outputFileName2, 0)

        inputFileName3 = "tiger.psd"
        outputFileName3 = inputFileName3 + ".pdf"
        doc3 = PDFDoc()
        Convert.ToPdf(doc3, input_path + inputFileName3)
        doc3.Save(output_path + outputFileName3, 0)

        print("DCM, HEIC and PSD image conversion example")
    PDFNet.Terminate()


if __name__ == '__main__':
    main()

File: Samples/AdvancedImagingTest/GO/AdvancedImaging_test.go
//---------------------------------------------------------------------------------------
// Copyright (c) 2001-2025 by Apryse Software Inc. All Rights Reserved.
// Consult LICENSE.txt regarding license information.
//---------------------------------------------------------------------------------------

package main
import (
	"fmt"
	"testing"
	"flag"
	. "github.com/pdftron/pdftron-go/v2"
)

var licenseKey string
var modulePath string

func init() {
	flag.StringVar(&licenseKey, "license", "", "License key for Apryse SDK")
	flag.StringVar(&modulePath, "modulePath", "", "Module path for Apryse SDK")
}

// ---------------------------------------------------------------------------------------
// The following sample illustrates how to use Advanced Imaging module
// --------------------------------------------------------------------------------------

// Relative path to the folder containing the test files.
var inputPath = "../TestFiles/AdvancedImaging/"
var outputPath = "../TestFiles/Output/"

//---------------------------------------------------------------------------------------

func catch(err *error) {
	if r := recover(); r != nil {
		*err = fmt.Errorf("%v", r)
	}
}

//---------------------------------------------------------------------------------------

func DicomTest() (err error) {
	defer catch(&err)

	inputFileName := "xray.dcm"
	outputFileName := inputFileName + ".pdf"
	doc := NewPDFDoc()
	ConvertFromDICOM(doc, inputPath + inputFileName)
	doc.Save(outputPath + outputFileName, uint(SDFDocE_linearized))

	return nil
}

//---------------------------------------------------------------------------------------

func HeicTest() (err error) {
	defer catch(&err)

	inputFileName := "jasper.heic"
	outputFileName := inputFileName + ".pdf"
	doc := NewPDFDoc()
	ConvertToPdf(doc, inputPath + inputFileName)
	doc.Save(outputPath + outputFileName, uint(SDFDocE_linearized))

	return nil
}

//---------------------------------------------------------------------------------------

func PsdTest() (err error) {
	defer catch(&err)

	inputFileName := "tiger.psd"
	outputFileName := inputFileName + ".pdf"
	doc := NewPDFDoc()
	ConvertToPdf(doc, inputPath + inputFileName)
	doc.Save(outputPath + outputFileName, uint(SDFDocE_linearized))

	return nil
}

//---------------------------------------------------------------------------------------

func TestAdvancedImagingModule(t *testing.T) {
	// The first step in every application using PDFNet is to initialize the 
	// library. The library is usually initialized only once, but calling 
	// Initialize() multiple times is also fine.
	PDFNetInitialize(licenseKey)

	//-----------------------------------------------------------------------------------

	PDFNetAddResourceSearchPath(modulePath)

	// Test if the add-on is installed
	if !AdvancedImagingModuleIsModuleAvailable() {
		fmt.Println("")
		fmt.Println("Unable to run AdvancedImagingTest: Apryse SDK AdvancedImaging module not available.")
		fmt.Println("-----------------------------------------------------------------------------")
		fmt.Println("The AdvancedImaging module is an optional add-on, available for download")
		fmt.Println("at https://docs.apryse.com/documentation/core/info/modules/. If you have already")
		fmt.Println("downloaded this module, ensure that the SDK is able to find the required files")
		fmt.Println("using the PDFNetAddResourceSearchPath() function.")
		fmt.Println("")
	} else {

		err := DicomTest()
		if err != nil {
			fmt.Println(fmt.Errorf("Unable to convert DICOM test file, error: %s", err))
		}

		err = HeicTest()
		if err != nil {
			fmt.Println(fmt.Errorf("Unable to convert the HEIC test file, error: %s", err))
		}

		err = PsdTest()
		if err != nil {
			fmt.Println(fmt.Errorf("Unable to convert the PSD test file, error: %s", err))
		}
		fmt.Println("Done.")
	}

	PDFNetTerminate()
}
